OpenRA.Mods.Common/Traits/Player/GameMode.cs

```diff
@@ -0,0 +1,57 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Defines a game mode and the condition that is being granted when it is active. Use this to enable",
+		"different game modes/VictoryCondition traits on the same map. Attach this to the Player or World actor.")]
+	class GameModeInfo : ITraitInfo
+	{
+		[FieldLoader.Require]
+		[Desc("Internal name used for this game mode.")]
+		public readonly string InternalName = null;
+
+		[FieldLoader.Require]
+		[Desc("Name of the game mode as shown in the UI.")]
+		public readonly string Name = null;
+
+		[FieldLoader.Require]
+		[GrantedConditionReference]
+		[Desc("The Condition being granted when this game mode is active.")]
+		public readonly string Condition = null;
+
+		public object Create(ActorInitializer init) { return new GameMode(this); }
+	}
+
+	class GameMode
+	{
+		public readonly GameModeInfo Info;
+
+		public GameMode(GameModeInfo info)
+		{
+			Info = info;
+		}
+	}
+
+	// GameModeInit is used to restrict Actors to certain game modes, eg. only showing KotH flags on the koth game mode.
+	// An empty value is interpreted to mean that the actor should appear in all game modes.
+	// Matches against the InternalName field of a GameMode.
+	public class GameModesInit : IActorInit<string[]>
```

Comment:
I suggested a very similar (but more general) Init in https://github.com/IceReaper/KKnD/issues/22.  Would be good to at least consider this here, but otherwise / at least I think we should expose this init in the new editor actor properties panel.